components/backends/vllm/src/dynamo/vllm/handlers.py

```diff
@@ -181,6 +198,12 @@ async def generate(self, request):
         # Generate only 1 token in prefill
         try:
             async for res in gen:
+                if context.is_stopped() or context.is_killed():
```

Comment:
should we have a comined is_cancelled that checks both ? also double check that we need both - not sure when the difference is material - maybe for logging?

Reply:
synced offline - the is_stopped() and is_killed() are one-to-one mapping to the Rust code. The is_stopped() signals graceful cancellation while is_killed() signals a hard stop.

Reply:
Filed a ticket followup on combining to is_cancelled(): DIS-567